spring-cloud-starter-dataflow-metrics-collector/src/main/java/org/springframework/cloud/dataflow/metrics/collector/MetricsAggregator.java

```diff
@@ -41,7 +37,11 @@ public MetricsAggregator(ApplicationMetricsService service) {
 
 	@StreamListener(Sink.INPUT)
 	public void receive(ApplicationMetrics metrics) {
-		this.service.add(metrics);
+		if (metrics.getProperties().get(ApplicationMetrics.APPLICATION_GUID) != null
+				&& metrics.getProperties().get(ApplicationMetrics.APPLICATION_NAME) != null
+				&& metrics.getProperties().get(ApplicationMetrics.STREAM_NAME) != null) {
+			this.service.add(metrics);
+		}
```

Comment:
It might be good idea to log on debug level that why we're skipping this metric. info level might possibly be too chatty.